pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -1104,6 +1104,9 @@ type ResourceTemplate struct {
 	// FailureCondition is a label selector expression which describes the conditions
 	// of the k8s resource in which the step was considered failed
 	FailureCondition string `json:"failureCondition,omitempty" protobuf:"bytes,6,opt,name=failureCondition"`
+
+	// IgnoreNotFound tells kubectl not to return an error if the resource is not found. It defaults to false
+	IgnoreNotFound bool `json:"ignoreNotFound,omitempty" protobuf:"varint,7,opt,name=ignoreNotFound"`
```

Comment:
I understand that we use `--ignore-not-found` when talking to `kubectl`, however from the perspective of Argo this feature provides a silent failure mode.

Perhaps it would be more clear if this was named `silent` or `silentFail`?

Reply:
@simster7 I would like to challenge the idea that this is a silent failure mode.  Any user who has need of this is not trying to make their workflows fail silently (if so, they would just use exithandlers and respond to the failure).  For the users who need this, not found is a completely valid, non failure response: Is this resource in the cluster? No.

If I create the workflow:
```
resource:
  silentFail: true
```
and the workflow step fails for me because the API server times out, I will be utterly confused.

If you must change the feature name to include the word failure to scare away other uses, I would suggest something like 'notFoundShouldFail' (with a default of true) so that users who include `notFoundShouldFail: false` will be able to make logical sense of their workflows

Reply:
Sure, fair point. `ignoreNotFound` is more explicit in this case.